lib/iaas/GcpClient.js

```diff
@@ -0,0 +1,200 @@
+'use strict';
+const _ = require('lodash');
+const Promise = require('bluebird');
+const GcpStorage = require('@google-cloud/storage');
+const logger = require('../logger');
+const errors = require('../errors');
+const utils = require('../utils');
+const ComputeClient = require('./ComputeClient');
+const BaseCloudClient = require('./BaseCloudClient');
+const NotFound = errors.NotFound;
+const Unauthorized = errors.Unauthorized;
+const Forbidden = errors.Forbidden;
+
+class GcpClient extends BaseCloudClient {
+  constructor(settings) {
+    super(settings);
+    this.constructor.validateParams(_.chain(this.settings).value());
+    this.storageClient = this.constructor.createStorageClient(_
+      .chain(this.settings)
+      .omit('name')
+      .set('provider', this.provider)
+      .value()
+    );
+    this.computeClient = this.constructor.createComputeClient(_
+      .chain(this.settings)
+      .set('provider', this.provider)
+      .value()
+    );
+  }
+
+  getContainer(container) {
+    if (arguments.length < 1) {
+      container = this.containerName;
+    }
+
+    return this.storageClient
+      .bucket(container)
+      .get();
+  }
+
+  list(container, options) {
+    if (arguments.length < 2) {
+      options = container;
+      container = this.containerName;
+    }
+    const prefix = options ? options.prefix : null;
```

Comment:
if options is defined but prefix is undefined, you would be setting prefix to undefined as against to null as is set when options is undefined. What should be the value of prefix if its missing? is it null or undefined? Fix same default value.

Reply:
I do not want to set prefix if null or undefined. Change the query to check both options and options.prefix.